pydantic_ai_slim/pydantic_ai/_parts_manager.py

```diff
@@ -120,43 +163,83 @@ def handle_text_delta(
+            return
 
         if existing_text_part_and_index is None:
             # This is a workaround for models that emit `<think>\n</think>\n\n` or an empty text part ahead of tool calls (e.g. Ollama + Qwen3),
             # which we don't want to end up treating as a final result when using `run_stream` with `str` a valid `output_type`.
             if ignore_leading_whitespace and (len(content) == 0 or content.isspace()):
-                return None
+                return
 
             # There is no existing text part that should be updated, so create a new one
-            new_part_index = len(self._parts)
             part = TextPart(content=content, id=id, provider_details=provider_details)
-            if vendor_part_id is not None:
-                self._vendor_id_to_part_index[vendor_part_id] = new_part_index
-            self._parts.append(part)
-            return PartStartEvent(index=new_part_index, part=part)
+            new_part_index = self._append_part(part, vendor_part_id)
+            yield PartStartEvent(index=new_part_index, part=part)
         else:
             # Update the existing TextPart with the new content delta
             existing_text_part, part_index = existing_text_part_and_index
             part_delta = TextPartDelta(content_delta=content, provider_details=provider_details)
             self._parts[part_index] = part_delta.apply(existing_text_part)
-            return PartDeltaEvent(index=part_index, delta=part_delta)
+            yield PartDeltaEvent(index=part_index, delta=part_delta)
+
+    def _update_raw_content(
+        self,
+        part: ThinkingPart,
+        part_index: int,
+        raw_content_delta: str,
+        raw_content_index: int,
+    ) -> ThinkingPart:
+        """Update raw_content in provider_details and return updated part."""
+        existing_details = dict(part.provider_details or {})
+        raw_content_list = list(existing_details.get('raw_content', []))
+        while len(raw_content_list) <= raw_content_index:
+            raw_content_list.append('')
+        raw_content_list[raw_content_index] += raw_content_delta
+        existing_details['raw_content'] = raw_content_list
+        updated = ThinkingPart(
+            content=part.content,
+            id=part.id,
+            signature=part.signature,
+            provider_name=part.provider_name,
+            provider_details=existing_details,
+        )
+        self._parts[part_index] = updated
+        return updated
+
+    def _find_raw_content_only_part(self, id: str) -> tuple[ThinkingPart, int] | None:
+        """Find a ThinkingPart that has only raw_content (no summary content yet).
+
+        Handles streaming scenarios where raw content arrives before
+        summaries. In such cases, raw content creates a ThinkingPart with empty content and
+        raw_content in provider_details. When summaries arrive later with a different
+        vendor_part_id, we need to find and reuse that existing part.
```

Comment:
Why would they have a different `vendor_part_id`? I think if we use the `item.id` as the `vendor_part_id`, the summaries will come in with the same ID as the raw reasoning, so we can just update the `content` for the existing `ThinkingPart`, if `summary_index == 0`. For higher summary index, we can put it in the `vendor_part_id` so they get new parts